src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -540,6 +603,11 @@ function _applyRenderWhitespace(lineContent: string, len: number, continuesWithW
 			isInWhitespace = false;
 		}
 
+		// If rendering whitespace on selection, check that the charIndex falls within a selection
+		if (isInWhitespace && selections) {
+			isInWhitespace = !!currentSelection && currentSelection.startOffset <= charIndex && currentSelection.endOffset > charIndex;
```

Comment:
nitpick: small style inconsistency, use here `currentSelection && ... ` instead of `!!currentSelection && ... `

Reply:
removing the `!!` causes a typescript error since `isInWhitespace` should be a boolean, so will leave as is